File: Projects/Dagger/Projects/Project-2/app/src/main/java/com/droid/code/ui/MainActivity.java
package com.droid.code.ui;

import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.droid.code.MyApplication;
import com.droid.code.R;
import com.droid.code.di.components.DaggerApplicationActivityComponent;
import com.droid.code.di.modules.ActivityModule;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject
    public MainViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DaggerApplicationActivityComponent
                .builder()
                .activityModule(new ActivityModule(this))
                .applicationComponent(((MyApplication)getApplication()).applicationComponent)
                .build().inject(this);

        Toast.makeText(this,viewModel.getSomeData(),Toast.LENGTH_LONG).show();
    }
}

File: Projects/Dagger/Projects/Project-5-DaggerInKotlin/app/src/main/java/com/demo/kotlin/sample/ui/main/MainActivity.kt
package com.demo.kotlin.sample.ui.main

import android.os.Bundle
import android.widget.TextView

import com.demo.kotlin.sample.MyApplication
import com.demo.kotlin.sample.R
import com.demo.kotlin.sample.di.component.DaggerActivityComponent
import com.demo.kotlin.sample.di.module.ActivityModule
import com.demo.kotlin.sample.ui.home.HomeFragment

import javax.inject.Inject

import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var viewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        getDependencies()
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val tvData = findViewById<TextView>(R.id.tv_message)
        tvData.text = viewModel.someData

        addHomeFragment()
    }

    private fun addHomeFragment() {
        if (supportFragmentManager.findFragmentByTag(HomeFragment.TAG) == null) {
            supportFragmentManager
                    .beginTransaction()
                    .add(R.id.container_fragment, HomeFragment.newInstance(), HomeFragment.TAG)
                    .commit()
        }
    }

    private fun getDependencies() {
        DaggerActivityComponent
                .builder()
                .applicationComponent((application as MyApplication).applicationComponent)
                .activityModule(ActivityModule(this))
                .build()
                .inject(this)
    }
}
